### PR TITLE
Fix some issues detected after commit #e3c9b6d

### PR DESCRIPTION
**Pull request by pablozg**
After check the changes made by perexg the add as previously recorded don't work properly.

Issues:

After select a recording as previously recorder is moved to "removed recordings", but if exist a new event of the recordings will recorded again.

After a restart of tvheadend the recordings added as "previously recorded" disappear from "removed recordings"

The recordings added as previously recorded show a status of "unknow status (109)".

The changes fix all issues and change the icon to show the same as "Completed OK".

### CODE REVIEW

File: src/dvr/dvr_db.c [hunk 1]
```diff
@@ -157,7 +157,7 @@ int dvr_entry_is_finished(dvr_entry_t *entry, int flags)
   int success = entry->de_sched_state == DVR_COMPLETED;
 
   if (success && entry->de_last_error != SM_CODE_FORCE_OK)
-      success = entry->de_last_error == SM_CODE_OK &&
+      success = (entry->de_last_error == SM_CODE_OK || entry->de_last_error == SM_CODE_PREVIOUSLY_RECORDED) &&
                 entry->de_data_errors < DVR_MAX_DATA_ERRORS;
```
Comment (pablozg) [inline src/dvr/dvr_db.c]: This prevent the rerecord issue.

File: src/dvr/dvr_db.c [hunk 2]
```diff
@@ -658,10 +658,11 @@ dvr_entry_status(dvr_entry_t *de)
     switch(de->de_last_error) {
       case SM_CODE_INVALID_TARGET:
         return N_("File not created");
+      case SM_CODE_PREVIOUSLY_RECORDED:
+        return N_("Previously Recorded");
       case SM_CODE_USER_ACCESS:
```
Comment (pablozg) [inline src/dvr/dvr_db.c]: Fix the unknow status

File: src/dvr/dvr_db.c [hunk 3]
```diff
@@ -711,6 +712,8 @@ dvr_entry_schedstatus(dvr_entry_t *de)
     rerecord = de->de_dont_rerecord ? 0 : dvr_entry_get_rerecord_errors(de);
     if(rerecord && (de->de_errors || de->de_data_errors > rerecord))
       s = "completedRerecord";
+    if(de->de_last_error == SM_CODE_PREVIOUSLY_RECORDED)
+      s = "completed";
     break;
```
Comment (pablozg) [inline src/dvr/dvr_db.c]: Change the icon to show the same as a "Completed OK" recording.

File: src/dvr/dvr_db.c [hunk 4]
```diff
@@ -4222,6 +4225,7 @@ dvr_entry_set_prevrec(dvr_entry_t *de, int cmd)
     de->de_dont_rerecord = 1;
     de->de_file_removed = 1;
     dvr_entry_completed(de, SM_CODE_PREVIOUSLY_RECORDED);
+    idnode_changed(&de->de_id);
   } else {
```
Comment (pablozg) [inline src/dvr/dvr_db.c]: Avoid the disappearance from "deleted recordings"

**Comment by perexg** [comment]
Thanks for your contribution. Fixes are in 98e4e70c35526972be49d4247920aabe6e8bf2e7, 39a4be55bffa4a264ea6ec73662f1f69c44c1c4a.

**Pull request closed without merging**